packages/context/test/acceptance/creating-and-resolving-bindings.ts

```diff
@@ -34,6 +34,26 @@ describe('Context bindings - Creating and resolving bindings', () => {
         }
       });
     });
+
+    context('with type information', () => {
+      it('infers correct type when getting the value', async () => {
+        const key = new BindingKey<string>('foo');
+        ctx.bind(key).to('value');
+        const value = await ctx.get(key);
+        // The following line is accessing a String property as a way
+        // of verifying the value type at compile time
+        expect(value.length).to.equal(5);
+      });
+
```

Comment:
Could we also get a test case for a custom type?

Reply:
What kind of a custom type would you like to test?

Reply:
I was thinking just any random classes, an empty one would be fine.